app/web/src/features/dashboard/CommunityBrowser.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  Divider,
+  Link as MuiLink,
+  List,
+  ListItem,
+  ListItemText,
+} from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import {
+  useCommunity,
+  useListSubCommunities,
+} from "features/communities/hooks";
+import { LOAD_MORE } from "features/dashboard/constants";
+import { Community } from "proto/communities_pb";
+import { ReactNode, useState } from "react";
+import { Link } from "react-router-dom";
+import { routeToCommunity } from "routes";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexDirection: "row",
+    "& > * + *": {
+      marginInlineStart: theme.spacing(2),
+    },
+  },
+  list: {
+    minWidth: "10rem",
+  },
+  loader: {
+    margin: "auto",
+  },
+  selected: {
+    fontWeight: "bold",
+  },
+}));
+
+export default function CommunityBrowser() {
+  const classes = useStyles();
+  const [selected, setSelected] = useState<Community.AsObject[]>([]);
+
+  const globalCommunityQuery = useCommunity(1);
+
+  //react-query doesn't have useInfiniteQueries
+  //as a workaround, cache query results
+  //and only show "Load more" for last column
+  const query = useListSubCommunities(
+    selected?.[selected.length - 1]?.communityId || 1
+  );
+  const [cachedQueryResults, setCachedQueryResults] = useState<
+    {
+      data: Community.AsObject[];
+      hasMore?: boolean;
+    }[]
+  >([]);
+
+  const handleClick = (community: Community.AsObject, level: number) => {
+    if (level === selected.length) {
```

Comment:
@darrenvong Level here is the level that was clicked on, so this if block adds the last level to the cache if it was clicked, otherwise an earlier level was clicked so it slices the cache to that level (or actually, one lower)

Reply:
Ohh so it's using the cached results to display previous levels and React Query's cache for the final level, so that you can fetch more communities at that final level if it has more than the default number of communities in one page?